Planes plugin reactivation

gzotti opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

### Description
<!--- Please include a summary of the change and which issue is fixed. -->
<!--- Please also include relevant motivation and context. -->
<!--- List any dependencies that are required for this change. -->

Recently, we had a discussion about tracking airplane traffic, similar to satellites (https://groups.google.com/g/stellarium/c/F4Z7lTI9C6M). Alexander recalled that in 2013, a plugin to track planes based on the ADS-B system was under development by @bi0ha2ard. Unfortunately it was never merged, and the current developers have no ADS-B hardware to test or update anything in this plugin. 

I only rebased this 10-year old code on top of current master (2023-03-08) and added enough updates so that it compiles with Qt5.15.2 and Qt6.4 and then added some obvious updates, e.g. use the StelProperty system, and harmonized the color selection. 

It may be possible to extend it based on e.g. data from FlightRadar24, by somebody more familiar with airplane tracking. However I cannot say currently what has to be changed. My time is too limited to continue on that, but maybe somebody else wants to take over?

This is a branch in the main repository. A merge to master is obviously only useful if the plugin demonstrably does something useful. If "conflict" warnings appear, they are usually solved quickly by a rebase from master at the onset of further work.  


Fixes # (issue)

### Screenshots (if appropriate):

### Type of change
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] This change requires a documentation update

### How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

**Test Configuration**:
* Qt version: 5.15.2, 6.4
* Operating system: Windows 11
* Graphics Card: (Geforce, but irrelevant)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (header file)
- [ ] I have updated the respective chapter in the Stellarium User Guide
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

bi0ha2ard committed:
initial commit (rebase 2023 and post-26.1)

bi0ha2ard committed:
cleanup
comments
C++11 build

bi0ha2ard committed:
more cleanup
small fixes

bi0ha2ard committed:
small fixes for getData() and ADSB

bi0ha2ard committed:
fix string in gui

bi0ha2ard committed:
make colour user configurable
load Planes on startup by default

bi0ha2ard committed:
store colour in settings as single string

bi0ha2ard committed:
remove unneeded class

bi0ha2ard committed:
Fix crash while connectiong database while old worker thread was still running

bi0ha2ard committed:
Bump version number to 0.1.0 and add it to the plugin info

bi0ha2ard committed:
Attempt to reconnect on connection loss setting

gzotti committed:
post-rebase fixes
- fixed immediate API changes (StelPosition, StelCore)
- adhere to new API for single-button plugin with right-click action
- add virtual methods where base classes had new abstract methods
- mark overrides where needed
- some things only commented out. Should be cleaned up ASAP.
- Plugin does not load! Needs fix ASAP...

gzotti committed:
Cosmetics
- Fixed missing/excessive headers
- declared another Q_ENUM
- also renamed an enum

alex-w commented:
> I only rebased this 10-year old code on top of current master (2023-03-08) and added enough updates so that it compiles, at least with Qt5.15.2. However, the plugin does not appear in the list of available plugins, so I obviously have overlooked something important. I hope Alexander can add this missing line of code/config (?), and then we can update it further to also compile with Qt6 and add the obvious updates, e.g. use the StelProperty system.

But it compiled and loaded without problems (Qt5)

![stellarium-001](https://user-images.githubusercontent.com/88731/224018017-bf000e6d-e8e4-497f-b868-20c383dd7d63.png)
![stellarium-002](https://user-images.githubusercontent.com/88731/224018073-e8ebd8d3-9883-4470-9d97-208e3fbfbd54.png)

gzotti commented:
Aargh, yes. Too many "start me" links here - I kept launching my Qt6-based build... :-o OK, I fix the description.

bi0ha2ard commented:
> Unfortunately it was never merged, and the current developers have no ADS-B hardware to test or update anything in this plugin.

Hi, I have some test data from back when I was developing this plugin. In case it's useful for testing, I've uploaded it here: https://tocsick.net/stellarium/adsb_test_data.7z

alex-w commented:
![stellarium-001](https://user-images.githubusercontent.com/88731/224127344-e86e4ac6-ee22-49a1-ada5-49f5bc56ea62.png)

@bi0ha2ard could you share info about ADS-B device?

gzotti commented:
@alex-w do you see a tool button? I can still only call up the config dialog via Config menu/plugins tab.
I now tried to modernize the UI and interface utilizing StelProperties. 
@bi0ha2ard thanks for the data, certainly useful to at least restore old functionality. (However I don't see anything :-( 

I see in the logfile:
Parsed  "D:/Stellarium/modules/Planes/adsb_test_data/Test Data/BaseStation Recording File/20131114-114822.bst" , contained  395  flights.
Ignored  1673  lines.
Is that OK? 

What was the reason to stop development, was it stuck in finalisation, or just not merged because "nobody" has such receiver? We would definitely need a page or two of instructions for the User Guide.

bi0ha2ard commented:
> ![stellarium-001](https://user-images.githubusercontent.com/88731/224127344-e86e4ac6-ee22-49a1-ada5-49f5bc56ea62.png)
> 
> @bi0ha2ard could you share info about ADS-B device?

This seems to be the manual for it: https://www.kinetic.co.uk/DownloadFiles/Basestation_183/BaseStationReferenceManual-v3-2-1.pdf (the website seems defunct otherwise?)

> @bi0ha2ard thanks for the data, certainly useful to at least restore old functionality. (However I don't see anything :-(

To see anything, your location must be set to Würzburg, Germany, and you need to set the time to the range for which data is available. I don't remember what the "ignoring" means, could just be that the data is in the past.

> What was the reason to stop development, was it stuck in finalisation, or just not merged because "nobody" has such receiver? We would definitely need a page or two of instructions for the User Guide.

This was developed for my Bachelor's thesis and deployed at the Uni department. They were running the build I handed in with their receiver hardware there. I don't know if further development happened afterwards.

alex-w commented:
> @alex-w do you see a tool button? I can still only call up the config dialog via Config menu/plugins tab. I now tried to modernize the UI and interface utilizing StelProperties. 

Yes, but the button is very small, because original texture is not very good ;)

alex-w commented:
@bi0ha2ard thanks for the info! Right now I have RTL-SDR receiver and in theory I can getting the data at 1090 MHz

gzotti commented:
Ah, that button! OK. --- Ahhh, and finally I see the traces! Cool!

gzotti committed:
Post-rebase simplifications
- pulled properties into Planes class
- simplified dialog
- added upscaled icon (preliminary!)
- cleanup

gzotti commented:
OK, a first round of updates is finished, and it creates useful output with the test file. Now someone please create some solution with online access to real-world current flight data! :-)

gzotti committed:
Fixes for Qt6

alex-w committed:
InfoBox fixes

alex-w committed:
Update PlanesDialog.ui

gzotti commented:
Congrats!
However, we should fix #3115 and maybe #3014 before weekend. Not sure about the others. I am pretty much occupied, though.

alex-w committed:
Fix GUI

alex-w committed:
cosmetic fixes

gzotti commented:
> @bi0ha2ard did you tested the code with real hardware?

He has described above, 
>This was developed for my Bachelor's thesis and deployed at the Uni department. They were running the build I handed in with their receiver hardware there. I don't know if further development happened afterwards.

alex-w committed:
Fix cmake

alex-w committed:
Fix cmake

gzotti committed:
Cosmetics
- Codefactor issues
- Update from Qt syntax recommendations

gzotti committed:
GUI updates

gzotti committed:
Search updates

gzotti committed:
Dissolved class FlightMgr

- integrated methods of Planes friend class into Planes
